lib/ash/actions/read/read.ex

```diff
@@ -1543,7 +1543,7 @@ defmodule Ash.Actions.Read do
             (opts[:page][:count] != false and action.pagination.countable == :by_default)) do
       with {:ok, filter} <-
              filter_with_related(
-               query_before_pagination,
+               query,
```

Comment:
I believe that here we still need to pass in the `query_before_pagination`. Was this changed due to an issue?